plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -84,43 +157,28 @@ String getProjectId()
 
     Iterable<Dataset> listDatasets(String projectId)
     {
-        Iterator<Dataset> datasets = bigQuery.listDatasets(projectId).iterateAll().iterator();
-        return () -> Iterators.transform(datasets, this::addDataSetMappingIfNeeded);
+        return bigQuery.listDatasets(projectId).iterateAll();
     }
 
-    Iterable<Table> listTables(DatasetId datasetId, TableDefinition.Type... types)
+    Iterable<Table> listTables(DatasetId remoteDatasetId, TableDefinition.Type... types)
```

Comment:
`datasetId` was probably OK

Reply:
It's package-private. And I changed to make caller be careful otherwise the call-site will start throwing exceptions because the dataset might not be all-lowercase.

I don't feel strongly about this though.

Reply:
so we have all-lowercase and translated DatasetId objects?

would it be possible to make sure the DatasetId object is always created from a translated (remote-ized) name?

Reply:
In the current code it's not possible without introducing a wrapper over DatasetId/TableId or changing all call-sites to pass Strings and the receiver be responsible for creating the DatasetId/TableId objects (which is what I've done in the follow-up series).

Ideally no class other than BigQueryClient should use any models from BigQuery SDK. Everything on the Trino end (mainly BigQueryMetadata) should happen on Strings and the only entrypoint to any BigQuery specific things should be in BigQueryClient.